packages/lodestar/src/sync/regular/oneRangeAhead/processor.ts

```diff
@@ -0,0 +1,68 @@
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {Root, SignedBeaconBlock} from "@chainsafe/lodestar-types";
+import {ILogger} from "@chainsafe/lodestar-utils";
+import {AbortSignal} from "abort-controller";
+import {IRegularSyncModules} from "..";
+import {ChainEvent, IBeaconChain} from "../../../chain";
+import {BlockError, BlockErrorCode} from "../../../chain/errors";
+import {sortBlocks} from "../../utils";
+import {IBlockRangeProcessor} from "./interface";
+
+/**
+ * Process a block range until complete.
+ */
+export class BlockRangeProcessor implements IBlockRangeProcessor {
+  protected readonly config: IBeaconConfig;
+  private readonly chain: IBeaconChain;
+  private readonly logger: ILogger;
+  private target: {root: Root; resolve: Function; signal: AbortSignal} | null;
+  constructor(modules: Pick<IRegularSyncModules, "config" | "chain" | "logger">) {
+    this.config = modules.config;
+    this.chain = modules.chain;
+    this.logger = modules.logger;
+    this.target = null;
+  }
+
+  public async start(): Promise<void> {
+    this.chain.emitter.on(ChainEvent.block, this.onProcessedBlock);
+    this.chain.emitter.on(ChainEvent.errorBlock, this.onErrorBlock);
+  }
+
+  public async stop(): Promise<void> {
+    this.chain.emitter.removeListener(ChainEvent.block, this.onProcessedBlock);
+    this.chain.emitter.removeListener(ChainEvent.errorBlock, this.onErrorBlock);
+  }
+
+  /**
+   * Main method.
+   */
+  public async processUntilComplete(blocks: SignedBeaconBlock[], signal: AbortSignal): Promise<void> {
+    if (!blocks || !blocks.length) return;
+    await new Promise((resolve) => {
+      const sortedBlocks = sortBlocks(blocks);
+      this.logger.info("Imported blocks for slots: " + blocks.map((block) => block.message.slot).join(","));
```

Comment:
The logger should handle the formatting you, whenever possible give it a message + JSON data